cmd/agent/common/misconfig/mounts.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package misconfig
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+func procMount() error {
+	groups, err := os.Getgroups()
+	if err != nil {
+		return fmt.Errorf("failed to get process groups: %v", err)
+	}
+	return checkProcMountHidePid("/host/proc/1/mounts", groups)
+}
+
+func checkProcMountHidePid(path string, groups []int) error {
+	file, err := os.Open(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil
+		}
```

Comment:
Why do we want to be silent when `/host/proc/1/mounts` doesn’t exist ?
If it doesn’t exist, there must be a misconfiguration that should be raised to the user.
I suppose it’s for not raising an error when the agent isn’t running inside a container and `/host/proc` doesn’t exist.
I think that even when the agent isn’t running inside a container, it makes sense to check for the `hidepid` option on `/proc` (instead of `/host/proc`).
So, I would be in favour of detecting whether we should check `/proc/mounts` or `/host/proc/1/mounts` depending on the context and to always raise an error.

Reply:
The issue is more with environments where we don't have any meaningful `/proc`, like Fargate.

Though I think `config.Datadog.GetString("container_proc_root")` returns `/proc` in this case, which should make it work there too.